# 📦 Export for Web Dashboard

Final notebook to prepare all assets for the GitHub Pages web dashboard.

In [1]:
import shutil
from pathlib import Path
import json

RESULTS = Path('../results')
WEBAPP = Path('../webapp/assets')
WEBAPP_DATA = WEBAPP / 'data'
WEBAPP_FIGURES = WEBAPP / 'figures'

# Create directories
WEBAPP_DATA.mkdir(parents=True, exist_ok=True)
WEBAPP_FIGURES.mkdir(parents=True, exist_ok=True)

print("📁 Directories ready:")
print(f"   Data: {WEBAPP_DATA}")
print(f"   Figures: {WEBAPP_FIGURES}")

📁 Directories ready:
   Data: ..\webapp\assets\data
   Figures: ..\webapp\assets\figures


In [2]:
# Copy figures
figures_src = RESULTS / 'figures'

print("\n📊 Copying figures...")
copied_count = 0
for fig in figures_src.glob('*.png'):
    shutil.copy(fig, WEBAPP_FIGURES)
    print(f"   ✅ {fig.name}")
    copied_count += 1

print(f"\n📦 Copied {copied_count} figures to {WEBAPP_FIGURES}")


📊 Copying figures...
   ✅ attention_training_curves.png
   ✅ attribution_comparison.png
   ✅ budget_reallocation.png
   ✅ cac_shift.png
   ✅ campaign_analysis.png
   ✅ campaign_type_analysis.png
   ✅ channel_distribution.png
   ✅ conversion_by_channel.png
   ✅ first_last_touch.png
   ✅ journey_analysis.png
   ✅ lstm_training_curves.png
   ✅ rule_based_attribution.png
   ✅ sample_attribution.png

📦 Copied 13 figures to ..\webapp\assets\figures


In [3]:
# Verify data export exists
data_file = WEBAPP_DATA / 'attribution_data.json'

if data_file.exists():
    with open(data_file, 'r') as f:
        data = json.load(f)
    print("\n📋 Data export verified:")
    print(f"   ✅ {data_file.name}")
    print(f"   • Attention AUC: {data['metrics']['attention_auc']:.4f}")
    print(f"   • Improvement: {data['metrics']['improvement_pct']:.1f}%")
    print(f"   • Channels: {len(data['attribution']['channels'])}")
else:
    print("\n⚠️ Data file not found! Please run 06_cac_comparison.ipynb first.")


📋 Data export verified:
   ✅ attribution_data.json
   • Attention AUC: 0.9208
   • Improvement: 31.5%
   • Channels: 10


In [4]:
# List all webapp files
webapp_root = Path('../webapp')

print("\n📁 Webapp Structure:")
for item in sorted(webapp_root.rglob('*')):
    if item.is_file():
        rel_path = item.relative_to(webapp_root)
        size_kb = item.stat().st_size / 1024
        print(f"   {rel_path} ({size_kb:.1f} KB)")


📁 Webapp Structure:
   assets\data\attribution_data.json (2.6 KB)
   assets\figures\attention_training_curves.png (87.2 KB)
   assets\figures\attribution_comparison.png (127.1 KB)
   assets\figures\budget_reallocation.png (104.5 KB)
   assets\figures\cac_shift.png (89.5 KB)
   assets\figures\campaign_analysis.png (109.9 KB)
   assets\figures\campaign_type_analysis.png (59.9 KB)
   assets\figures\channel_distribution.png (103.5 KB)
   assets\figures\conversion_by_channel.png (55.0 KB)
   assets\figures\first_last_touch.png (57.8 KB)
   assets\figures\journey_analysis.png (98.2 KB)
   assets\figures\lstm_training_curves.png (91.7 KB)
   assets\figures\rule_based_attribution.png (80.6 KB)
   assets\figures\sample_attribution.png (30.8 KB)
   assets\model\.gitkeep (0.0 KB)
   css\style.css (18.9 KB)
   index.html (14.1 KB)
   js\app.js (12.5 KB)


In [5]:
print("""
============================================================
DEPLOYMENT INSTRUCTIONS
============================================================

1. Push to GitHub:
   git add .
   git commit -m "Marketing Attribution with Deep Learning"
   git push origin main

2. Enable GitHub Pages:
   - Go to Settings → Pages
   - Source: Deploy from a branch
   - Branch: main, folder: /webapp
   - Save

3. Your site will be live at:
   https://yourusername.github.io/marketing-attribution-dl/

============================================================
LOCAL TESTING
============================================================

To test locally, run:
   cd webapp
   python -m http.server 8000

Then open: http://localhost:8000
""")

print("\n✅ All assets exported successfully!")


DEPLOYMENT INSTRUCTIONS

1. Push to GitHub:
   git add .
   git commit -m "Marketing Attribution with Deep Learning"
   git push origin main

2. Enable GitHub Pages:
   - Go to Settings → Pages
   - Source: Deploy from a branch
   - Branch: main, folder: /webapp
   - Save

3. Your site will be live at:
   https://yourusername.github.io/marketing-attribution-dl/

LOCAL TESTING

To test locally, run:
   cd webapp
   python -m http.server 8000

Then open: http://localhost:8000


✅ All assets exported successfully!


---

## 🎉 Project Complete!

### Summary of Notebooks:

| Notebook | Purpose | Status |
|----------|---------|--------|
| 01_data_exploration | Explore Criteo dataset | ✅ |
| 02_data_preprocessing | Create user journeys | ✅ |
| 03_rule_based_models | Last-Touch, Linear, etc. | ✅ |
| 04_lstm_attribution | LSTM model training | ✅ |
| 05_attention_model | Attention-based model | ✅ |
| 06_cac_comparison | CAC shift analysis ⭐ | ✅ |
| 07_export_for_web | Web dashboard export | ✅ |

### Key Findings:

- Deep Learning achieves **~92% AUC** vs **~70% baseline**
- Many campaigns are **undervalued** by Last-Touch attribution
- CAC shifts of up to **23%** when using proper attribution
- Attention weights provide **interpretable** attribution scores